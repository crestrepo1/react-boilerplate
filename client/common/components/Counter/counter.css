._ {
    margin: 0 auto;
    text-align: center;
}

.button {
    appearance: none;
    background: #2980B9;
    border: 0;
    border-radius: 5px;
    color: #FFF;
    margin: 10px;
    padding: 10px;
}
