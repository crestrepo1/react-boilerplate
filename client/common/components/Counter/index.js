import React, { useContext } from 'react';
import { observer } from 'mobx-react-lite';
import { CounterStoreContext } from '../../stores/CounterStore';

import styles from './counter.css';

const CounterView = observer(() => {
    const counterStore = useContext(CounterStoreContext);
    return (
        <div className={styles._}>
            <h2>{counterStore.count}</h2>
            <button className={styles.button} onClick={() => counterStore.increaseCount()}>Increase</button>
            <button className={styles.button} onClick={() => counterStore.decreaseCount()}>Decrease</button>
        </div>
    );
});

export default CounterView;
