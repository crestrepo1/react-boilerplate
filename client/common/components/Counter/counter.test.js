import React from 'react';
import { shallow } from 'enzyme';
import Counter from './index.js';

describe('MyComponent', () => {
    it('should render correctly', () => {
        const component = shallow(<Counter />);
        expect(component).toMatchSnapshot();
    });
});
