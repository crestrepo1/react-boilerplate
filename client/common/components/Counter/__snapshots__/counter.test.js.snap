// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User View Component should render correctly 1`] = `
<div
  className="_"
>
  <h2>
    0
  </h2>
  <button
    className="button"
    onClick={[Function]}
  >
    Increase
  </button>
  <button
    className="button"
    onClick={[Function]}
  >
    Decrease
  </button>
</div>
`;
